src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,359 @@
+        {
+            if (string.IsNullOrWhiteSpace(languageName))
+            {
+                throw new LanguageExtensibilityException($"Invalid language name. name: {languageName}");
+            }
+            Dictionary<string, object> parameters = new Dictionary<string, object>();
+            ExecuteNonQuery(connection, GetDropScript(languageName), parameters);
+        }
+
+        /// <summary>
+        /// Returns the status of external languages in a connection
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        private List<ExternalLanguage> GetLanguages(IDbConnection connection, string languageName = null)
+        {
+            Dictionary<int, ExternalLanguage> dic = new Dictionary<int, ExternalLanguage>();
+            try
+            {
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = languageName != null ? GetLanguageScript : GetAllScript;
+                    if (languageName != null)
+                    {
+                        var parameter = command.CreateParameter();
+                        parameter.ParameterName = $"@{LanguageNameParamName}";
+                        parameter.Value = languageName;
+                        command.Parameters.Add(parameter);
+                    }
+                    using (IDataReader reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            int id = reader.GetInt32(0);
+                            string name = reader.GetString(1);
+                            string createdDate = reader.IsDBNull(2) ? string.Empty : reader.GetDateTime(2).ToString();
+                            string owner = reader.IsDBNull(3) ? string.Empty : reader.GetString(3);
+                            string extentionFile = string.Empty;//reader.IsDBNull(4) ? string.Empty : reader.GetString(4);
+                            string extentionFileName = reader.IsDBNull(5) ? string.Empty : reader.GetString(5);
+                            string platform = reader.IsDBNull(6) ? string.Empty : reader.GetString(6);
+                            string parameters = reader.IsDBNull(7) ? string.Empty : reader.GetString(7);
+                            string envVariables = reader.IsDBNull(8) ? string.Empty : reader.GetString(8);
+                            if (!dic.ContainsKey(id))
+                            {
+                                dic.Add(id, new ExternalLanguage
+                                {
+                                    Name = name,
+                                    Owner = owner,
+                                    CreatedDate = createdDate,
+                                    Contents = new List<ExternalLanguageContent>()
```

Comment:
Why not just init contents here?

Reply:
this block should be outside of if 

---
In reply to: [374369743](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r374369743) [](ancestors = 374369743)